### PR TITLE
Split token parsing

### PR DESCRIPTION
**Pull request by obtitus**
Hi, new to golang and tried to create an application around kiva.org api: https://www.kiva.org/build as a learning project.

I got an error with the calls to `url.ParseQuery` because it does not like semicolon. There might be much better ways to solve this, but attached:

1. Splits up the `RequestToken` and `AccessToken` into `RequestTokenString` and `AccessTokenString` respectively, I am fine if this is the only change accepted as this at least allows the user to parse the output themselves.
2. Adds new `RequestTokenJSON` and `AccessTokenJSON`, which is a drop in replacement which assumes the returned body is .json. I used:
```
RequestTokenURL: https://api.kivaws.org/oauth/request_token.json
AuthorizeURL: https://www.kiva.org/oauth/authorize?client_id=<client_id>&response_type=code&oauth_callback=oob&scope=access
AccessTokenURL: https://api.kivaws.org/oauth/access_token.json
```
where <client_id> is my application named registered on kiva.org.

example body format (not a valid secret):
```
body = {"oauth_token":"Syi-Bjn9QXEQ9j5otA30fbiJRgj1234;app-name","oauth_token_secret":"eYz0dPtCTlI23rWwY6I4GsdEp-1234","oauth_callback_confirmed":"true"}
```
resulted in error from url.ParseQuery:
```
Unable to request token: invalid semicolon separator in query
```

Open to changes and feedback

### CODE REVIEW

**Comment by dghubble** [comment]
Temproary and access token responses are included in the HTTP response body using the "application/x-www-form-urlencoded" content type, which uses URL encoding. Semicolons in these parameters or JSON format are not part of the OAuth1 spec.

**Comment by obtitus** [comment]
I can't really do anything about the kiva api. Would you be more willing to accept just the first commit so that the user can parse any non-standard response on their own?

File: config.go [hunk 1]
```diff
@@ -105,6 +118,40 @@ func (c *Config) RequestToken() (requestToken, requestSecret string, err error)
...
+		return "", "", err
+	}
+
+	// Parse JSON
+	var values requestResponse
+	err = json.Unmarshal([]byte(body), &values)
+	if err != nil {
+		return "", "", err
+	}
+	fmt.Printf("values = %v\n", values)
```
Comment (obtitus) [inline config.go:143]: will remove these debug prints, but lets decide on the way forward first.

**Comment by dghubble** [comment]
Unfortunately I'm not really interested in behaviors outside the oauth1 spec. You may wish to reach out to the provider or maintain your own fork with the custom logic you need

**Pull request closed without merging**